test/logger_backend_test.exs

```diff
@@ -205,68 +211,165 @@ defmodule Sentry.LoggerBackendTest do
+      _json = Jason.decode!(body)
+      send(pid, "API called")
+      Plug.Conn.resp(conn, 200, ~s<{"id": "340"}>)
+    end)
+
+    capture_log(fn ->
+      :hackney.get("http://127.0.0.1:8003/error_route", [], "", [])
+      assert_receive "API called"
+      assert_receive "API called"
+    end)
+  after
+    :ok = Plug.Cowboy.shutdown(Sentry.ExamplePlugApplication.HTTP)
+    Logger.configure_backend(Sentry.LoggerBackend, excluded_domains: [:cowboy])
+  end
+
+  test "includes Logger.metadata for keys configured to be included" do
+    Logger.configure_backend(Sentry.LoggerBackend, metadata: [:string, :number, :map, :list])
+    bypass = Bypass.open()
+    Process.flag(:trap_exit, true)
+    pid = self()
+
+    Bypass.expect(bypass, fn conn ->
+      {:ok, body, conn} = Plug.Conn.read_body(conn)
+      json = Jason.decode!(body)
+      assert get_in(json, ["extra", "logger_metadata", "string"]) == "string"
+      assert get_in(json, ["extra", "logger_metadata", "map"]) == %{"a" => "b"}
+      assert get_in(json, ["extra", "logger_metadata", "list"]) == [1, 2, 3]
+      assert get_in(json, ["extra", "logger_metadata", "number"]) == 43
+      send(pid, "API called")
+      Plug.Conn.resp(conn, 200, ~s<{"id": "340"}>)
+    end)
+
+    modify_env(:sentry, dsn: "http://public:secret@localhost:#{bypass.port}/1")
+
+    capture_log(fn ->
+      {:ok, pid} = Sentry.TestGenServer.start_link(pid)
+      Sentry.TestGenServer.add_logger_metadata(pid, :string, "string")
+      Sentry.TestGenServer.add_logger_metadata(pid, :number, 43)
+      Sentry.TestGenServer.add_logger_metadata(pid, :map, %{a: "b"})
+      Sentry.TestGenServer.add_logger_metadata(pid, :list, [1, 2, 3])
+      Sentry.TestGenServer.invalid_function(pid)
+      assert_receive "terminating"
+      assert_receive "API called"
+    end)
+  end
+
+  test "does not include Logger.metadata when disabled" do
+    Logger.configure_backend(Sentry.LoggerBackend, metadata: [])
+    bypass = Bypass.open()
+    Process.flag(:trap_exit, true)
+    pid = self()
+
+    Bypass.expect(bypass, fn conn ->
+      {:ok, body, conn} = Plug.Conn.read_body(conn)
+      json = Jason.decode!(body)
+      assert get_in(json, ["extra", "logger_metadata"]) == %{}
+      send(pid, "API called")
+      Plug.Conn.resp(conn, 200, ~s<{"id": "340"}>)
+    end)
+
+    modify_env(:sentry, dsn: "http://public:secret@localhost:#{bypass.port}/1")
+
+    capture_log(fn ->
+      {:ok, pid} = Sentry.TestGenServer.start_link(pid)
+      Sentry.TestGenServer.add_logger_metadata(pid, :string, "string")
+      Sentry.TestGenServer.add_logger_metadata(pid, :atom, :atom)
+      Sentry.TestGenServer.add_logger_metadata(pid, :number, 43)
+      Sentry.TestGenServer.add_logger_metadata(pid, :list, [])
+      Sentry.TestGenServer.invalid_function(pid)
+      assert_receive "terminating"
+      assert_receive "API called"
+    end)
+  end
+
+  test "sends all messages if capture_log_messages is true" do
```

Comment:
Let's add a test that the level is respected. For example, Logger.debug is never sent. :)

Reply:
Done!